packages/evm/src/exceptions.ts

```diff
@@ -35,6 +37,39 @@ export enum ERROR {
   INVALID_COMMITMENT = 'kzg commitment does not match versioned hash',
   INVALID_INPUTS = 'kzg inputs invalid',
   INVALID_PROOF = 'kzg proof invalid',
+
+  // EOF
+  INVALID_EOF_MAGIC = 'invalid EOF magic',
+  INVALID_EOF_VERSION = 'invalid EOF version',
+
+  INVALID_KIND_TYPE = 'invalid EOF KIND_TYPE',
+  INVALID_TYPE_COUNT = 'invalid EOF Type count',
+
+  INVALID_HEADER_SIZE = 'invalid EOF header size',
+
+  INVALID_KIND_CODE = 'invalid EOF KIND_CODE',
+  INVALID_CODE_COUNT = 'invalid EOF Code count',
+  INVALID_CODE_COUNT_TYPES = 'invalid EOF Code count, should be Type count divided by 4',
+  INVALID_CODE_SIZE = 'invalid EOF Code size',
+
+  INVALID_KIND_DATA = 'invalid EOF KIND_DATA',
+
+  INVALID_TERMINATOR_BYTE = 'invalid EOF: invalid header terminator byte',
+
+  INVALID_CONTAINER_COUNT = 'invalid EOF Container count',
+  INVALID_CONTAINER_SIZE = 'invalid EOF Container size',
+
+  INVALID_EOF_INPUTS = 'invalid EOF type section: too much inputs',
+  INVALID_EOF_OUTPUTS = 'invalid EOF type section: too much outputs',
+  INVALID_EOF_MAX_STACK_HEIGHT = 'invalid EOF type section: too high max stack height',
+
+  INVALID_EOF_CODE_OPCODE_UNDEFINED = 'invalid EOF bytecode: undefined opcode found',
+
+  INVALID_EOF_DATA_SECTION_SIZE = 'invalid EOF data section size: data section too large',
+
+  INVALID_EOF_FIRST_CODE_SECTION_TYPE = 'invalid EOF: first code section must have 0 inputs and should have 0x80 (non-returning) outputs',
+
+  INVALID_EOF_FINAL_SECTION_OPCODE = 'invalid EOF: code section does not have a valid final (terminating) instruction opcode',
```

Comment:
Just throwing in one-off questions and not doing in the form a full review yet: was is the difference/distinction between the errors here and the ones in `eof/errors.ts`? This partly looks redundant on first look?

Reply:
I have moved all errors, except INVALID EOF CONTAINER inside the EOF errors. The reason that the prior stays in exceptions is that it is an EvmError and therefore needs to stay there. (This error is also caused by EIP-3541, it is CREATE/CREATE2 trying to deploy EOF containers)